core/unix/signal.c

```diff
@@ -712,8 +717,9 @@ create_clone_record(dcontext_t *dcontext, reg_t *app_thread_xsp)
              * cl_args->stack. But we expect the highest (non-inclusive)
              * in the clone record's app_thread_xsp.
              */
-            record->app_thread_xsp = app_clone_args->stack + app_clone_args->stack_size;
-            record->clone_flags = app_clone_args->flags;
+            record->app_thread_xsp = ((clone3_syscall_args_t *)app_clone_args)->stack +
```

Comment:
These field accesses should check the size to confirm they are present.

Reply:
I don't think we need to worry about fields like `stack`, `stack_size` and `flags` being absent. I cannot imagine `clone3` without these essential args. Also, the [old struct that Assad observed](https://github.com/DynamoRIO/dynamorio/issues/5131#issuecomment-958971904) also has these fields.

Reply:
If in future DR starts accessing some new field added by the kernel, we should add the size check for those accesses. But it doesn't seem necessary for these ever-present fields.

Reply:
The earliest version of clone_args, in v5.3: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=8f6ccf6159aed1f04c6d179f61f6fb2691261e84.

It has all fields till tls.

Reply:
Now we return `EINVAL` if the clone args size passed by the app is less than the required.